src/UI/Art/SpriteArtist/UnitSpriteSet.java

```diff
@@ -169,13 +172,26 @@ public void drawUnitIcons(Graphics g, Commander activeCO, Unit u, int drawX, int
           null);
     }
     
-    // Draw a Commander-defined symbol, if applicable.
-    char symbol = activeCO.getSymbol(u);
-    if( '\0' != symbol) // optimization; the null char is invisible
+    // Collect all the Commanders' desired symbols
+    // This is ripe for optimization, possibly, but tracking the data and keeping it updated is probably more effort than it's worth.
+    Map<Commander, Character> coSymbols = new HashMap<Commander, Character>();
+    for( Commander co : COs )
     {
-      BufferedImage num = SpriteLibrary.getColoredMapTextSprites(activeCO.myColor).get(symbol);
+      char symbol = co.getCustomSymbol(u);
+      if( '\0' != symbol ) // null char is our sentry value
+      {
+        coSymbols.put(co, Character.toUpperCase(symbol));
+      }
+    }
+
+    // Draw one of them, based on our animation index
+    if( !coSymbols.isEmpty() )
+    {
+      @SuppressWarnings("unchecked") // Can't create an array of generic Entries to pass into toArray(), so we have to cast after the fact
```

Comment:
You could avoid the warning by just grabbing the nth entry from the KeySet, and then accessing the value using the key.

Also, I'm not sure a map is really the best option here. You could just drop the commanders who have symbols into an array, and then make a second call to `getCustomSymbol` on the one at your chosen index.

Reply:
Since it's a set and sets are unordered, I wasn't able to.

....and your second paragraph is why we have code review.